database/src/selector.rs

```diff
@@ -250,20 +252,22 @@ impl BenchmarkQuery for CompileBenchmarkQuery {
     ) -> Result<Vec<SeriesResponse<Self::TestCase, StatisticSeries>>, String> {
         let mut statistic_descriptions: Vec<_> = index
             .compile_statistic_descriptions()
-            .filter(|(&(b, p, s, backend, m), _)| {
+            .filter(|(&(b, p, s, backend, m, ref compiler_target), _)| {
                 self.benchmark.matches(b)
                     && self.profile.matches(p)
                     && self.scenario.matches(s)
                     && self.backend.matches(backend)
                     && self.metric.matches(m)
+                    && self.compiler_target.matches(compiler_target.to_string())
```

Comment:
```suggestion
                    && self.compiler_target.matches(&**compiler_target)
```
Bit of a mouthful, but at least we don't need to clone the string, as this part is a bit perf. sensitive.